### PR TITLE
infoschema: add some annotation for columnInfo #34745

### LINKED ISSUE #34745: Add some annotation for `columnInfo`
## Enhancement
Add some annotation for `columnInfo` in https://github.com/pingcap/tidb/blob/master/infoschema/tables.go#L281-L290 which are important information for column but have no explanation for each attribution.

### PR DESCRIPTION
**Pull request by fanrenhoo**
Signed-off-by: fanrenhoo <fanrenhoo@sina.com>

### What problem does this PR solve?

Issue Number: close #34745

Problem Summary: Add some annotation for columnInfo of INFORMATION_SCHEMA db

### Check List

Tests 
No
- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects
No
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation
No
- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Add some annotation for columnInfo of INFORMATION_SCHEMA db
```

### CODE REVIEW

**Review by Lloyd-Pottiger — APPROVED** [review]
LGTM

File: infoschema/tables.go [hunk 1]
```diff
@@ -278,14 +278,19 @@ var tableIDMap = map[string]int64{
...
+// columnInfo represents the basic column information of all kinds of INFORMATION_SCHEMA tables
 type columnInfo struct {
-	name      string
-	tp        byte
-	size      int
-	decimal   int
-	flag      uint
-	deflt     interface{}
-	comment   string
+	name string
```
Comment (xhebox) [inline infoschema/tables.go:283]: Let us explain every field.

File: infoschema/tables.go [hunk 2]
```diff
@@ -278,14 +278,19 @@ var tableIDMap = map[string]int64{
...
+	// tp represent column type
+	tp      byte
+	size    int
+	decimal int
+	// flag represent NotNull, Unsigned, PriKey flags etc.
+	flag uint
+	// deflt represent default value
+	deflt   interface{}
+	comment string
+	// enumElems represent elements when the column type is enum
```
Comment (xhebox) [inline infoschema/tables.go:293]: Maybe more detailed, e.g. `enumElems` are all possible literal string values of an enum column

**Review by xhebox — COMMENTED** [review]
Anyway, I think you could link this to https://github.com/pingcap/tidb/blob/a96285a640ec36e5686f56dfce7e78006c467ab6/parser/types/field_type.go#L40-L50.

`FieldType` is more essential. It may be a good idea to explain `FieldType` and say `columnInfo` is highly similar to `FieldType`.

File: infoschema/tables.go (hunk 2 above)
Comment (fanrenhoo) [reply infoschema/tables.go]: ok, THKS

**Comment by fanrenhoo** [comment]
Hi, done with the revise follow suggestion, please check. Cause FieldType is not all same as columnInfo, so I chose to add annotations both sides.

File: infoschema/tables.go [hunk 3]
```diff
@@ -278,14 +278,23 @@ var tableIDMap = map[string]int64{
...
+// columnInfo represents the basic column information of all kinds of INFORMATION_SCHEMA tables
 type columnInfo struct {
-	name      string
-	tp        byte
-	size      int
-	decimal   int
-	flag      uint
-	deflt     interface{}
-	comment   string
+	// represent column name
```
Comment (xhebox) [inline infoschema/tables.go:283]: Comment should start with `name`, i.e. the name of the struct field.

File: infoschema/tables.go [hunk 4]
```diff
@@ -278,14 +278,23 @@ var tableIDMap = map[string]int64{
...
+	name string
+	// tp represent column type
```
Comment (xhebox) [inline infoschema/tables.go:285]: You could just say `tp is column type`, it is more concise.

**Review by xhebox — COMMENTED** [review]
Rest LGTM

**Comment by xhebox** [comment]
> Hi, done with the revise follow suggestion, please check. Cause FieldType is not all same as columnInfo, so I chose to add annotations both sides.

You could re-request reviewers to review by clicking the button under the right-side reviewers panel. It will trigger an notification, too, without leaving any comment. You don't need to comment every time, and leaving comments as replies to reviewers or self-reviewing are better, since reviews are attached to code closely.

File: infoschema/tables.go (hunk 4 above)
Comment (fanrenhoo) [reply infoschema/tables.go:285]: OK

Comment (fanrenhoo) [reply infoschema/tables.go]: Done. please recheck. THKS

File: parser/types/field_type.go [hunk 5]
```diff
@@ -39,11 +39,17 @@ var (
...
 // FieldType records field type information.
 type FieldType struct {
-	tp      byte
-	flag    uint
-	flen    int
+	// tp is type of the field
+	tp byte
+	// flag represent NotNull, Unsigned, PriKey flags etc.
+	flag uint
+	// represent size of bytes of the field
```
Comment (Defined2014) [inline parser/types/field_type.go:46]:
```suggestion
	// flen represent size of bytes of the field
```

File: parser/types/field_type.go [hunk 6]
```diff
@@ -39,11 +39,17 @@ var (
...
+	flen int
+	// represent decimal length of the field
```
Comment (Defined2014) [inline parser/types/field_type.go:48]:
```suggestion
	// decimal represent decimal length of the field
```

File: parser/types/field_type.go [hunk 7]
```diff
@@ -39,11 +39,17 @@ var (
...
 	decimal int
+	// represent character set
```
Comment (Defined2014) [inline parser/types/field_type.go:50]:
```suggestion
	// charset represent character set
```

File: parser/types/field_type.go [hunk 8]
```diff
@@ -39,11 +39,17 @@ var (
...
 	charset string
+	// represent collate rules of the charset
```
Comment (Defined2014) [inline parser/types/field_type.go:52]:
```suggestion
	// collate represent collate rules of the charset
```

**Review by Defined2014 — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

**Pull request merged**